### PR TITLE
Replaced hasOwnProperty with hasOwn in code examples

### PR DESCRIPTION
**Pull request by maurer2**
#### Summary
Hello, I replaced `hasOwnProperty` with `hasOwn` as suggested here https://github.com/orgs/mdn/discussions/143 (number 6) in various pages.
I left it in place in `en-us/web/javascript/reference/global_objects/object/hasown/index.md` and `en-us/web/javascript/reference/global_objects/object/hasownproperty/index.md` as those pages explicitly mention the differences between the two functions. Some other pages already had examples with both functions so I left them untouched was well.

#### Motivation
Improves legibility of code samples.

#### Supporting details
Discussion: https://github.com/orgs/mdn/discussions/143

This PR…
- [x] Rewrites (or significantly expands) a document

### CODE REVIEW

**Comment by teoli2003** [comment]
Thanks a lot! Your PR is very valuable.

Before doing a formal review, thanks to your PR, I raised a question [there](https://github.com/orgs/mdn/discussions/143#discussioncomment-3118634). I think there is an extra pattern we should fix at the same time.

Once it is settled, we'll move forward with this PR.

File: files/en-us/web/javascript/reference/statements/var/index.md [hunk 1]
```diff
@@ -95,7 +95,7 @@ operator on a global variable.
 ```js
 'use strict';
 var x = 1;
-globalThis.hasOwnProperty('x'); // true
+Object.hasOwn(globalThis, 'x') // true
```
Comment (zloirock) [inline files/en-us/web/javascript/reference/statements/var/index.md:98]: Missed `;` that's present in all the rest code of the example.

**Review by teoli2003 — CHANGES_REQUESTED** [review]
This looks good. I think there is one case where we can simplify further and get rid of `hasOwnProperty`/`hasOwn`, and simplify even more!

File: files/en-us/learn/tools_and_testing/cross_browser_testing/automated_testing/index.md [hunk 2]
```diff
@@ -450,7 +450,7 @@ Let's have a brief look at how we'd access the API using Node.js and [node-sauce
         myAccount.getJobs(function (err, jobs) {
           // Get a list of all your jobs
           for (let k in jobs) {
```
[the remark this answers is not in the document]
Comment (maurer2) [reply files/en-us/learn/tools_and_testing/cross_browser_testing/automated_testing/index.md]: Okay, make sense. Thanks. I implemented the change.

**Review by teoli2003 — APPROVED** [review]
👍

**Pull request merged**